### PR TITLE
Some changes in preparation for big "include everything but jujuignored" branch.

### PR DESCRIPTION
**Pull request by facundobatista**
Included here:

- put the 'hooks' dir as a constant

- introduced a 'relativise' helper, that will be used in the next branch 

- added a verification in the build command line: the entry point must be inside the project

### CODE REVIEW

File: charmcraft/commands/build.py [hunk 1]
```diff
@@ -79,6 +81,21 @@ def polite_exec(cmd):
     return retcode
 
 
+def relativise(src, dst):
```
Comment (justinmclark) [inline charmcraft/commands/build.py:84]: Could we just use `os.path.relpath()` here? Something like `os.path.relpath(dst, src.parent)` should return the same thing.

Comment (chipaca) [reply charmcraft/commands/build.py:84]: `os.path.relpath` falls flat on its face for most of these:

| `src` | `dst` | `relativise(src, dst)` | `os.path.relpath(dst, src.parent)` |
| --- | --- | -- | -- |
| `/tmp/foo/bar/src.txt` | `/tmp/foo/bar/dst.txt` | `dst.txt` | `dst.txt` |
| `/tmp/foo/bar/baz/src.txt` | `/tmp/foo/dst.txt` | `../../dst.txt` | `dst.txt` |
| `/tmp/foo/src.txt` | `/tmp/foo/bar/baz/dst.txt` | `bar/baz/dst.txt` | `dst.txt` |
| `/tmp/foo/bar/src.txt` | `/tmp/foo/baz/dst.txt` | `../baz/dst.txt` | `dst.txt` |
| `/tmp/foo/bar1/bar2/src.txt` | `/tmp/foo/baz1/baz2/baz3/dst.txt` | `../../baz1/baz2/baz3/dst.txt` | `dst.txt` |

File: charmcraft/commands/build.py [hunk 2]
```diff
@@ -79,6 +81,21 @@ def polite_exec(cmd):
...
+    src_rel_parts = []
+    dst_rel_parts = []
+    for src_part, dst_part in itertools.zip_longest(src.parent.parts, dst.parts):
+        if src_part != dst_part:
+            if src_part is not None:
+                src_rel_parts.append('..')
+            if dst_part is not None:
+                dst_rel_parts.append(dst_part)
+
+    link = pathlib.Path(os.path.join(*(src_rel_parts + dst_rel_parts)))
```
Comment (chipaca) [inline charmcraft/commands/build.py:95]: the `os.path.join` call seems spurious to me (all tests pass if I remove it)

also `*(a+b)` is the same as `*a, *b`, but that might be an aesthetic thing

File: charmcraft/commands/build.py (hunk 1 above)
Comment (justinmclark) [reply charmcraft/commands/build.py:84]: This is interesting to me. I didn't directly run the test cases, but I locally tested these 4 and they return the same result using `os.path.relpath`

e.g. 
```
>>> src = pathlib.Path("/tmp/foo/bar1/bar2/src.txt")
>>> dst = pathlib.Path("/tmp/foo/baz1/baz2/baz3/dst.txt")
>>> os.path.relpath(dst, src.parent)
'../../baz1/baz2/baz3/dst.txt'
```

Comment (chipaca) [reply charmcraft/commands/build.py:84]: I don't know what I did to get those results now. I can reproduce what you're doing there and get what you got. Sadly the script I used to create the table was in `/tmp` and I had to reboot :-(
But hey, this means less code \o/

**Pull request merged**